From dc317a8a6624ccaa3ebc635f6c28a03972cc4ae6 Mon Sep 17 00:00:00 2001
From: Przemyslaw Czarnowski <przemyslaw.hawrylewicz.czarnowski@intel.com>
Date: Mon, 30 May 2022 16:41:46 +0200
Subject: [PATCH] Combined two defect fixes

Upstream forced patches to be rearanged. Remaining two patches that has
to be moved after "Apply async dbus api".

Seal "NotConnected" usage

When media is disconnected, "NotConnected" field did not exist.
This change adds default value of this field for all code paths.

Tested:
curl -X GET https://some_addr/redfish/v1/Managers/bmc/VirtualMedia/Legacy
{
  "@odata.id": "/redfish/v1/Managers/bmc/VirtualMedia/Legacy",
  "@odata.type": "#VirtualMedia.v1_3_0.VirtualMedia",
  "Actions": {
    "#VirtualMedia.EjectMedia": {
      "target": "/redfish/v1/Managers/bmc/VirtualMedia/Legacy/Actions/VirtualMedia.EjectMedia"
    },
    "#VirtualMedia.InsertMedia": {
      "target": "/redfish/v1/Managers/bmc/VirtualMedia/Legacy/Actions/VirtualMedia.InsertMedia"
    }
  },
  "ConnectedVia": "NotConnected",
  "Id": "Legacy",
  "Inserted": false,
  "MediaTypes": [
    "CD",
    "USBStick"
  ],
  "Name": "Virtual Removable Media",
  "Oem": {
    "OpenBMC": {
      "@odata.type": "#OemVirtualMedia.v1_0_0.VirtualMedia"
    }
  },
  "TransferMethod": "Stream",
  "WriteProtected": true
}

Change-Id: I5e0789947e092907abe1bc353aebc07a4fb87274
Signed-off-by: Przemyslaw Czarnowski <przemyslaw.hawrylewicz.czarnowski@intel.com>

Rename doVmAction to doEjectAction

Change-Id: I69050daa01e0def44d6bfcf26c56551c4306bbd5
Signed-off-by: Przemyslaw Czarnowski <przemyslaw.hawrylewicz.czarnowski@intel.com>
---
 redfish-core/lib/virtual_media.hpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/redfish-core/lib/virtual_media.hpp b/redfish-core/lib/virtual_media.hpp
index e0ca63db..edd1b83c 100644
--- a/redfish-core/lib/virtual_media.hpp
+++ b/redfish-core/lib/virtual_media.hpp
@@ -269,6 +269,7 @@ inline nlohmann::json vmItemTemplate(const std::string& name,
         "/redfish/v1/Managers/{}/VirtualMedia/{}", name, resName);
 
     item["@odata.type"] = "#VirtualMedia.v1_3_0.VirtualMedia";
+    item["ConnectedVia"] = "NotConnected";
     item["Name"] = "Virtual Removable Media";
     item["Id"] = resName;
     item["WriteProtected"] = true;
-- 
2.25.1

